packages/squiggle-lang/src/rescript/GenericDist/GenericDist_Types.res

```diff
@@ -0,0 +1,92 @@
+type genericDist = [
+  | #PointSet(PointSetTypes.pointSetDist)
+  | #SampleSet(array<float>)
+  | #Symbolic(SymbolicDistTypes.symbolicDist)
+]
+
+type error =
+  | NotYetImplemented
+  | ImpossiblePath
+  | DistributionVerticalShiftIsInvalid
+  | Other(string)
+
+module Operation = {
+  type direction = [
+    | #Algebraic
+    | #Pointwise
+  ]
+
+  type arithmeticOperation = [
+    | #Add
+    | #Multiply
+    | #Subtract
+    | #Divide
+    | #Exponentiate
+    | #Log
+  ]
+
+  let arithmeticToFn = (arithmetic: arithmeticOperation) =>
+    switch arithmetic {
+    | #Add => \"+."
+    | #Multiply => \"*."
+    | #Subtract => \"-."
+    | #Exponentiate => \"**"
+    | #Divide => \"/."
+    | #Log => (a, b) => log(a) /. log(b)
+    }
+
+  type toFloat = [
+    | #Cdf(float)
+    | #Inv(float)
+    | #Mean
+    | #Pdf(float)
+    | #Sample
+  ]
+
+  type toDist = [
+    | #normalize
+    | #toPointSet
+    | #toSampleSet(int)
+    | #truncate(option<float>, option<float>)
+    | #consoleLog
+  ]
+
+  type toFloatArray = [
+    | #Sample(int)
+  ]
+
+  type fromDist = [
+    | #toFloat(toFloat)
+    | #toDist(toDist)
+    | #toDistCombination(direction, arithmeticOperation, [#Dist(genericDist) | #Float(float)])
+    | #toString
+  ]
+
+  type singleParamaterFunction = [
+    | #fromDist(fromDist)
+    | #fromFloat(fromDist)
+  ]
+
+  type genericFunctionCall = [
```

Comment:
hmmm items of type `genericFunctionCall` aren't actually _calls_. I think of _calls_ as the verb, the moment of the calling actually happening, not just the _info_ of what will be a call getting lined up and situated. Maybe `genericFunctionCallInfo`?

Reply:
Yea, this one is tricky. I'm not sure. Maybe something like, ``genericFunctionInputs`` or ``genericFunctionCallInputs``?

Reply:
`genericFunctionCallInfo` is still my favorite, also consider `genericFunctionWithInputs`